app/src/ui/app.tsx

```diff
@@ -186,6 +187,15 @@ export class App extends React.Component<IAppProps, IAppState> {
     return this.state.currentPopup !== null || this.state.errors.length > 0
   }
 
+  /**
+   * Returns a memoized instance of onPopupDismissed() bound to the
+   * passed popupType, so it can be used in render() without creating
+   * multiple instances when the component gets re-rendered.
+   */
+  private getOnPopupDismissedFn = (popupType: PopupType) => {
+    return () => this.onPopupDismissed(popupType)
+  }
+
```

Comment:
Should we drop this in favor of reintroducing it properly in the memoization PR?
```suggestion
```

Reply:
Without this check, in some situations the dismissal of this popup will cause other popups to get closed: I've just tested it and when merging the conflicts modal will get automatically closed after stashing the changes.

Anyways, I'm planning to send a PR with the memoization very soon, so in either case we'll fix the bug or the performance issue very soon 😃

Reply:
Ah, that's great context that I was missing. Sounds great, we can live with the performance issue for a brief period of time I think.